### PR TITLE
Add min and max duration validation to date range answers

### PR DESCRIPTION
**Pull request by adammahmood**
### What is the context of this PR?
Adds min and max duration validation to date range answers.  This enforces users in `eq-survey-runner` to enter a date within the specified duration period. 

### How to review 
- Ensure test pass
- Create a date range answer and add a min/max duration validation rule and ensure `eq-survey-runner` shows appropriate validation messages when trying to submit data outside of the min/max duration rules set

### CODE REVIEW

**Comment by adammahmood** [comment]
[Validation for date range - Min duration and max duration (9)](https://trello.com/c/Qr9lDfQw/407-validation-for-date-range-min-duration-and-max-duration-9)

File: eq-author-api/schema/typeDefs.js [hunk 1]
```diff
@@ -285,6 +292,18 @@ type LatestDateValidationRule implements ValidationRule {
   availableMetadata: [Metadata!]!
 }
 
+type MinDurationValidationRule implements ValidationRule {
+  id: ID!
+  enabled: Boolean!
+  offset: Duration!
```
Comment (tgandrews) [inline eq-author-api/schema/typeDefs.js]: I dislike using the an offset here. It isn't an offset. It is more like min or max value.

File: eq-author/src/components/Validation/Date/DatePreview.js [hunk 2]
```diff
@@ -45,9 +44,10 @@ const DatePreview = ({
   return (
     <React.Fragment>
       <div>
-        {`${value} ${unit.toLowerCase()} ${relativePosition.toLowerCase()}:`}
+        {`${value} ${unit.toLowerCase()} `}
+        {relativePosition && `${relativePosition.toLowerCase()}:`}
       </div>
-      <div>{rule[entityType]}</div>
+      <div>{entityType && rule[entityType]}</div>
```
Comment (tgandrews) [inline eq-author/src/components/Validation/Date/DatePreview.js]: Do we want an empty div when its not needed?

File: eq-author/src/components/Validation/Date/builder.js [hunk 3]
```diff
@@ -45,13 +45,13 @@ export const readToWriteMapper = outputKey => ({
   ...rest
 }) => ({
   id,
-  [outputKey]: {
+  [outputKey]: omitBy(isNil)({
```
Comment (tgandrews) [inline eq-author/src/components/Validation/Date/builder.js]: :+1: Nice pattern.

File: eq-author-api/schema/typeDefs.js (hunk 1 above)
Comment (adammahmood) [reply eq-author-api/schema/typeDefs.js]: It's definitely a `Duration` which we already described in our schema as:

```
type Duration {
  value: Int
  unit: DurationUnit!
}

enum DurationUnit {
  Days
  Months
  Years
}
```
What if we renamed `offset` to `duration`?

```
type MinDurationValidationRule implements ValidationRule {
  id: ID!
  enabled: Boolean!
  duration: Duration!
```

File: eq-author/src/components/Validation/Date/DatePreview.js (hunk 2 above)
Comment (adammahmood) [reply eq-author/src/components/Validation/Date/DatePreview.js]: No. Ideally, we should keep DOM elements to a minimum.  I'll rework this.

File: eq-author-api/schema/typeDefs.js (hunk 1 above)
Comment (tgandrews) [reply eq-author-api/schema/typeDefs.js]: :+1: Yes, that looks much better

**Pull request merged**